main.go

```diff
@@ -0,0 +1,189 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/google/go-github/v45/github"
+	"github.com/wandera/helm-github/helm"
+	"golang.org/x/oauth2"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	protocol      = "github"
+	githubHost    = "github.com"
+	indexFilename = "index.yaml"
+)
+
+var client *github.Client
+
+func init() {
+	token, err := loadGithubToken()
+	if err != nil {
+		log.Panic(err)
+	}
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: token},
+	)
+	tc := oauth2.NewClient(context.Background(), ts)
+	client = github.NewClient(tc)
+}
+
+func main() {
+	if env, ok := os.LookupEnv("HELMGITHUB_DEBUG_LOG"); ok {
+		t, err := os.OpenFile(env, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0o644)
+		if err != nil {
+			log.Panic(err)
+		}
+		defer t.Close()
+		log.SetOutput(t)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	if len(os.Args) == 5 {
+		uri := strings.TrimPrefix(os.Args[4], protocol+"://")
+		if strings.HasSuffix(uri, indexFilename) {
+			file, err := fetchIndexFile(ctx, uri)
+			if err != nil {
+				log.Panic(err)
+			}
+			file = switchProtocolToGithub(file)
+			file.SortEntries()
+			bytes, err := yaml.Marshal(&file)
+			if err != nil {
+				log.Panic(err)
+			}
+			fmt.Println(string(bytes))
+		} else {
+			rc, err := fetchArchive(ctx, uri)
+			if err != nil {
+				log.Panic(err)
+			}
+			defer rc.Close()
```

Comment:
when there is no archive found, Close() will be called on nil?

Reply:
Nope client's body will always be present(even empty one) unless method returns an error (in which case the flow ends early with panic)

Reply:
seems that practically cannot iff hcr tool uploads assets correctly (btw I was curious about line 140)